### PR TITLE
Check maximum length is not exceeded for incremental hashing.

### PR DESCRIPTION
**Pull request by msprotz**
This is an ABI-breaking change that makes `update` return an error code, which is either Success or MaximumLengthExceeded (new).

This also relieves the OCaml bindings from having to track this through an extra reference.

### CODE REVIEW

**Comment by msprotz** [comment]
Note that as a drive-by change I:
- fixed a notoriously flaky proof
- ditched those uu_ functions from EverCrypt_Error

**Review by victor-dumitrescu — APPROVED** [review]
The OCaml bindings part LGTM. Thanks!

File: bindings/ocaml/EverCrypt.mli [hunk 1]
```diff
@@ -114,6 +115,8 @@ module Hash : sig
     When using the streaming interface, the total number of bytes passed through {!update} must not exceed
     - 2{^61} for SHA-224, SHA-256, and the legacy algorithms
     - 2{^125} for SHA-384 and SHA-512
+    - 2{^64} for BLAKE2s
+    - 2{^128} for BLAKE2b
```
Comment (victor-dumitrescu) [inline bindings/ocaml/EverCrypt.mli:119]: I've added the mention of Blake2 to the streaming hash interface docstring

Comment (msprotz) [reply bindings/ocaml/EverCrypt.mli:119]: thanks! although right now the internal length counter is 64 bits so you won't be able to hash *that* much

I will fix this in a followup PR so no need to change the doc for now

Comment (victor-dumitrescu) [reply bindings/ocaml/EverCrypt.mli:119]: Yeah I remember this being the case, we were only checking up to 2^64 following the latest update to the bindings
```
assert (Z.lt !incr_len max_uint64);
```

**Review by sonmarcho — APPROVED** [review]

**Comment by msprotz** [comment]
@hacl-star/maintainers need an approving review on this, thanks!

**Review by R1kM — APPROVED** [review]

**Pull request merged**